### PR TITLE
Add Arm builds

### PR DESCRIPTION
**Pull request by mcbarton**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mcbarton** [comment]
@JohanMabille @SylvainCorlay can you review this PR for me please?

**Comment by mcbarton** [comment]
After https://github.com/conda-forge/cpp-argparse-feedstock/pull/5 was merged I was able to add a Linux Arm build, so this PR now covers both osx and Linux Arm builds.

**Comment by mcbarton** [comment]
@JohanMabille @SylvainCorlay ping

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ requirements:
     - xeus >=3.0.0,<4
     - xtl >=0.7.7,<0.8
     - pugixml
-    - cpp-argparse >=2.9,<3
+    - cpp-argparse >=2.9,<=3.0
```
Comment (SylvainCorlay) [inline recipe/meta.yaml:33]: I think that including 3.0 in the range is fine, but I would prefer using `<3.1` to allow for patch version upgrades.

Comment (SylvainCorlay) [inline recipe/meta.yaml]:
```suggestion
    - cpp-argparse >=2.9,<3.1
```

**Comment by SylvainCorlay** [comment]
@mcbarton I merged your PR after a couple of minor changes.

PS: I noticed that you opened the PR from the main branch. You may want to reset this branch to the new upstream/main.

**Pull request merged**